src/bb8_backup.py

```diff
@@ -0,0 +1,43 @@
+from subprocess import run
+
+targets = ["orderly"]
+finished_setup = False
+bb8_dir = "../montagu-bb8/bb8"
+
+
+def requires_bb8_setup(func):
+    def decorated(*args, **kwargs):
+        global finished_setup
+        if not finished_setup:
+            setup()
+        return func(*args, **kwargs)
+
+    return decorated
+
+
+def setup():
+    global finished_setup
+    template = "- Configuring and installing bb8 backup service with these " \
+               "targets: {}"
+    print(template.format(targets))
+    args = ["./setup.sh", "../config.json"] + targets
+    run(args, cwd=bb8_dir, check=True)
+    finished_setup = True
+
+
+@requires_bb8_setup
+def backup():
+    print("Performing bb8 backup")
+    run(["bb8", "backup"], check=True)
+
+
+@requires_bb8_setup
+def schedule():
+    print("Scheduling bb8 backup")
+    run("./schedule", cwd=bb8_dir, check=True)
```

Comment:
Does this mean we still need to deploy as root? What does the workflow look like if this step fails?

For barman, I allowed this part to fail and printed a message to run.  One way of managing that would be to add a flag to `schedule` that controls if it must work or not and printing the instructions there

Reply:
Yeah, currently, if there is anything that bb8 needs to do as root (which there will only sometimes be) I intend the deploy to fail and you to rerun it as root. I fear that a warning message will just get lost?